Source/Spray/SprayJet.cpp

```diff
@@ -9,81 +9,167 @@ SprayJet::SprayJet(const std::string& jet_name, const amrex::Geometry& geom)
+    ps.get("dpm_filename", m_FluentDPMFile);
+    std::ifstream is_dpm(m_FluentDPMFile, std::ios::in);
+    if (!is_dpm.good()) {
+      amrex::FileOpenFailed(m_FluentDPMFile);
+    } else {
+      long int dummy_line_no;
+      amrex::Real dummy_dpm_time;
+      amrex::Print() << "Using spray particle injection file: "
+                     << m_FluentDPMFile << "\n";
+      is_dpm >> m_dpm_num_of_time_instants;
+      is_dpm >> m_dpm_time_initial >> dummy_line_no;
+      for (long int i = 0; i < m_dpm_num_of_time_instants - 1; i++) {
+        is_dpm >> dummy_dpm_time >> dummy_line_no;
+      }
+      m_dpm_time_final = dummy_dpm_time;
+
+      amrex::Print() << "Total number of DPM time instants = "
+                     << m_dpm_num_of_time_instants << "\n";
+      amrex::Print() << "Spray injection initial time = " << m_dpm_time_initial
+                     << "\n";
+      amrex::Print() << "Spray injection final time = " << m_dpm_time_final
+                     << "\n";
+    }
+    is_dpm.close();
+    ps.query("m_override_inj_plane_dir", m_override_inj_plane_dir);
+    if (!(m_override_inj_plane_dir == -1 || m_override_inj_plane_dir == 0 ||
+          m_override_inj_plane_dir == 1 || m_override_inj_plane_dir == 2)) {
+      amrex::Abort("Override injection plane direction should be -1,0,1 or 2");
+    }
+    ps.query(".", m_rstrt_Fltdpmsim_from_nonFltDPMChckPointFile);
+    ps.query("m_override_inj_plane_loc", m_override_inj_plane_loc);
+    ps.query("is_dpm_periodic", is_dpm_periodic);
+
+    std::vector<amrex::Real> tmatrix(AMREX_SPACEDIM * AMREX_SPACEDIM);
+    std::vector<amrex::Real> dX_translation(AMREX_SPACEDIM);
+    ps.getarr("trans_matrix", tmatrix);
+    ps.getarr("translation", dX_translation);
+    for (int dir = 0; dir < AMREX_SPACEDIM * AMREX_SPACEDIM; ++dir) {
+      m_trans_matrix[dir] = tmatrix[dir];
+    }
+
+    for (int dir = 0; dir < AMREX_SPACEDIM; ++dir) {
+      m_translation[dir] = dX_translation[dir];
     }
-  }
-  ps.query("inject_ppp", m_numPPP);
-  // If a rate shape profile is generated at
-  // https://www.cmt.upv.es/#/ecn/download/InjectionRateGenerator/InjectionRateGenerator,
-  // it can be provided here for use
-  if (ps.contains("roi_file")) {
-    amrex::Real cd = 0.;
-    ps.get("discharge_coeff", cd);
-    std::string roi_file;
-    ps.get("roi_file", roi_file);
-    readROI(roi_file, rho_part, cd);
-    m_useROI = true;
-  } else {
     ps.query("start_time", m_startTime);
-    ps.query("end_time", m_endTime);
-    ps.get("jet_vel", m_jetVel);
-    m_maxJetVel = m_jetVel;
-    ps.get("mass_flow_rate", m_massFlow);
-    if (m_jetVel < 0. || m_massFlow < 0.) {
-      amrex::Abort("Jet u, T, and mdot must be greater than 0");
+    // ps.query("end_time", m_endTime);
+    m_endTime = verylargenum;
+
+    ps.get("initial_injection_dpm_time", m_dpm_time_initial_injection);
+    ps.get("initial_injection_flow_time", m_flow_time_initial_injection);
+    if (is_dpm_periodic) {
```

Comment:
What's going on here? It looks like this conditional isn't doing anything due to the commented out code?

Reply:
Removed the commented line which was used initially for debugging purposes